packages/stellar-axelar-std-derive/src/into_event.rs

```diff
@@ -55,9 +55,12 @@ pub fn into_event(input: &DeriveInput) -> proc_macro2::TokenStream {
             // Parse data from Val to the corresponding types,
             // and assign them to a variable with the same name as the struct field
             // E.g. let message = Message::try_from_val(env, &data.get(0));
-            // `data` is required to be a `Vec<Val>`
             let data = Vec::<Val>::try_from_val(env, &data)
```

Comment:
```suggestion
            let data = if singleton_data {
                Vec::<Val>::try_from_val(env, &data)
            } else {
                let mut vec = Vec::<Val>::new(env);
                vec.push_back(data);
                vec
            };
```
make it explicit instead. Something like
```rust
pub struct TransferEvent {
    pub from: Address,
    pub to: Address,
    #[singleton_data]
    pub amount: i128,
}
```
or
```rust
    #[singleton_data]
    #[data]
    pub amount: i128,
```

@nbayindirli